File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.cpp
#include <string>
#include <vector>
#include <numeric>

using namespace std;

int solution(vector<int> arr1, vector<int> arr2) {
    int sum1 = accumulate(arr1.begin(), arr1.end(), 0);
    int sum2 = accumulate(arr2.begin(), arr2.end(), 0);
    if (arr1.size() == arr2.size()){
        if(sum1 == sum2) return 0;
        else return sum1 > sum2 ? 1 : -1;
    } else return arr1.size() > arr2.size() ? 1 : -1;
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.js
function solution(arr1, arr2) {
    const sum1 = arr1.reduce((accumulator, n) => accumulator + n, 0);
    const sum2 = arr2.reduce((accumulator, n) => accumulator + n, 0);    

    if (arr1.length == arr2.length){
        if(sum1 == sum2) return 0;
        else return sum1 > sum2 ? 1 : -1;
    } else return arr1.length > arr2.length ? 1 : -1;
}